examples/interpreter-demo/backwards.html

```diff
@@ -0,0 +1,320 @@
+                <block type="math_arithmetic" inline="true">
+                  <field name="OP">MULTIPLY</field>
+                  <value name="A">
+                    <block type="variables_get">
+                      <field name="VAR">n</field>
+                    </block>
+                  </value>
+                  <value name="B">
+                    <block type="math_number">
+                      <field name="NUM">2</field>
+                    </block>
+                  </value>
+                </block>
+              </value>
+              <next>
+                <block type="text_print" id="print">
+                  <value name="TEXT">
+                    <block type="variables_get">
+                      <field name="VAR">n</field>
+                    </block>
+                  </value>
+                </block>
+              </next>
+            </block>
+          </statement>
+        </block>
+      </next>
+    </block>
+  </xml>
+
+  <script>
+    var demoWorkspace = Blockly.inject('blocklyDiv',
+        {media: 'https://unpkg.com/blockly/media/',
+         toolbox: document.getElementById('toolbox')});
+    Blockly.JavaScript.STATEMENT_PREFIX = 'highlightBlock(%1);\n';
+    Blockly.JavaScript.addReservedWords('highlightBlock');
+    Blockly.Xml.domToWorkspace(document.getElementById('startBlocks'),
+                               demoWorkspace);
+
+    var outputArea = document.getElementById('output');
+    var myInterpreter = null;
+
+    // Stack of serializations.  Each object contains three properties:
+    //   .json: A full serialization of the interpreter's state.
+    //   .delta: Instead of a full serialization, a delta from the next state.
+    //   .highlight: The ID of the block highlighted in this state.
+    var serializationStack = [];
+
+    // Global variable to keep track of the currently highlighted block.
+    var highlightedBlockId = '';
+
+    // The forward step button disables for 2s after program completion, then
+    // re-enables.  If the backwards step button is pressed during this time
+    // then cancel the scheduled re-enabling.
+    var disablePid = 0;
+
+    // Optionally, use the Diff Match Patch library to compress the stack.
+    var dmp;
+    if (typeof diff_match_patch === 'function') {
+      dmp = new diff_match_patch();
+      console.log('Using DMP for compression.');
+    } else {
+      console.warn('DMP not available, each step will cost ~300 KB.');
+    }
+
+    function initApi(interpreter, globalObject) {
+      // Add an API function for the alert() block, generated for "text_print" blocks.
+      var wrapper = function alert(text) {
+        text = arguments.length ? text : '';
+        outputArea.value += '\n' + text;
+      };
+      interpreter.setProperty(globalObject, 'alert',
+          interpreter.createNativeFunction(wrapper));
+
+      // Add an API function for the prompt() block.
+      var wrapper = function prompt(text) {
```

Comment:
Use different names for the wrappers (or include them inline) instead of redeclaring a var with the same name each time.